quickstart/java/redistest/src/main/java/example/demo/App.java

```diff
@@ -1,25 +1,78 @@
+    private static String host;
+    private static int port;
+    private static String password;
+    private static JedisPoolConfig config;
 
-        boolean useSsl = true;
+    // Should be called exactly once during app startup logic.
+    public static void initializeSettings(String host, int port, String password) {
+        App.host = host;
+        App.port = port;
+        App.password = password;
+    }
+
+    public static JedisPool getPoolInstance() {
+        if (pool == null) { // avoid synchronization lock if initialization has already happened
+            synchronized(staticLock) {
+                if (pool == null) { // don't re-initialize if another thread beat us to it.
+                    JedisPoolConfig poolConfig = getPoolConfig();
+                    boolean useSsl = port == 6380 ? true : false;
+                    int db = 0;
+                    pool = new JedisPool(poolConfig, host, port, Protocol.DEFAULT_TIMEOUT, password, db, useSsl);
+                }
+            }
+        }
+        return pool;
+    }
+
+    public static JedisPoolConfig getPoolConfig() {
+        if (config == null) {
+            JedisPoolConfig poolConfig = new JedisPoolConfig();
+
+            // Each thread trying to access Redis needs its own Jedis instance from the pool.
+            // Using too small a value here can lead to performance problems, too big and you have wasted resources.
+            int maxConnections = 200;
+            poolConfig.setMaxTotal(maxConnections);
+            poolConfig.setMaxIdle(maxConnections);
+
+            // Using "false" here will make it easier to debug when your maxTotal/minIdle/etc settings need adjusting.
+            // Setting it to "true" will result better behavior when unexpected load hits in production
+            poolConfig.setBlockWhenExhausted(true);
+
+            // How long to wait before throwing when pool is exhausted
+            poolConfig.setMaxWaitMillis(JedisPoolConfig.DEFAULT_MAX_WAIT_MILLIS);
+
+            // This controls the number of connections that should be maintained for bursts of load.
+            // Increase this value when you see pool.getResource() taking a long time to complete under burst scenarios
+            poolConfig.setMinIdle(50);
+
+            config = poolConfig;
+        }
+
+        return config;
+    }
+
+    public static void main( String[] args ) {
         String cacheHostname = System.getenv("REDISCACHEHOSTNAME");
         String cachekey = System.getenv("REDISCACHEKEY");
 
         // Connect to the Azure Cache for Redis over the TLS/SSL port using the key.
-        JedisShardInfo shardInfo = new JedisShardInfo(cacheHostname, 6380, useSsl);
-        shardInfo.setPassword(cachekey); /* Use your access key. */
-        Jedis jedis = new Jedis(shardInfo);      
+        initializeSettings(cacheHostname, 6380, cachekey);
+        JedisPool jedisPool = getPoolInstance();
+        Jedis jedis = jedisPool.getResource();
```

Comment:
In addition to `JedisPool`, the [gist](https://gist.github.com/JonCole/925630df72be1351b21440625ff2671f#file-redis-bestpractices-java-jedis-md) also recommends pipelining.

/cc @deepakverma for a little look, in case you have any recommendations or preferences about whether the quickstart should demonstrate pipelining or whether a non-quickstart sample would be a good fit for pipelining. Thanks in advance!

Relatedly, I think for StackExchange.Redis, the way to get pipelining is to call the async APIs. I was considering drafting a pull request for the .NET quickstarts to have a place to chat with the team about whether it would be good to steer developers toward the async APIs by default in the quickstarts or whether it would be better to leave those for a non-quickstart sample.